### DISCUSSION TITLE
Linux has more audio latency than windows

### DISCUSSION THREAD

**Discussion by Watenk** [ppy/osu#28558] [Q&A] [answered]
Hi, 
I just started playing osu lazer on linux but the audio latency is far worse than on windows while using the same pc. In windows i can play with a universal offset of 0ms but while using linux my offset needs to be around 67ms. I'm using the flatpack version (i also tried the AppImage but no difference in latency). Any idea what could cause this extra delay? I saw in another discussion that increasing the alsa buffer size could help but that didn't improve the latency at all.

**Comment by Stoppedpuma** [comment, marked as answer]
Please see [this PR](https://github.com/ppy/osu-framework/pull/6002), audio latency is a work in-progress on all platforms and linux will be stuck with using alsa configuration until this is merged.

> I'm using the flatpack version (i also tried the AppImage but no difference in latency)

Weird that the appimage is reproducing issues related to this as I've had issues in the past with the flatpak but not the appimage.

> Any idea what could cause this extra delay?

More information about your audio setup is needed to take any guesses.

> I saw in another discussion that increasing the alsa buffer size could help but that didn't improve the latency at all.

The instructions are for users with Pipewire installed, make sure you aren't using Pulseaudio if you're planning on using those.

**Reply by bdach** [reply]
> Please see [this PR](https://github.com/ppy/osu-framework/pull/6002)

Just to manage expectations: We've never explicitly committed to merging this. At best it's there because someone put it there and we don't care that much to reject it. It's a possible option but I can't say that it *will* 100% get merged.

**Comment by Watenk** [comment]
Thx for the fast answers. 
I installed pipewire (i tought i was already using it but apparently not) and the audio latency decreased by about 30ms. Still pretty high but alot better. Looking forward to the PR merge (if it gets merged).
